floodmodeller_api/toolbox/model_build/structure_log/structure_log.py

```diff
@@ -1,14 +1,46 @@
+import copy
 import csv
 
 from floodmodeller_api import DAT
 
 
+def extract_attrs(object_input, keys):
+    """
+    extract attributes and their values from a given object and return as a dict
+
+    No handling for where attributes dont exist but can ignore for when this is an issue, shouldnt be an issue here
+    """
+    return {key: getattr(object_input, key) for key in keys}
+
+
+def serialise_keys(old_dict):
+    """
+    dictionary keys must be unique; but in FM the node label isnt always unique
+    however the (label,type) pair will always be unique; so we have to use that as our key.
+    json doesnt like tuples as keys so we have to convert them first if we need to serialise our dictionary output.
+
+    Replaces tuple keys with string versions of themselves, wrapped in ()
+
+    """
+    new_dict = {}
+    for key, value in old_dict.items():
+        new_dict[f'({",".join(key)})'] = value
+
+    return new_dict
+
+
 class StructureLogBuilder:
-    def __init__(self, input_path, output_path) -> None:
+
+    def __init__(self, input_path="", output_path="") -> None:
```

Comment:
Why are the defaults for each argument `""` - is it helpful?

Reply:
This was to allow the testing scenarios and the checker stuff to initialise an instance of the SLB with blank input/output paths and 'manually feed' it, in all fairness None could have been better

Reply:
I think this is one where we should fix it as it's not just the internal workings of a class but its interface - could you please change the defaults to `None`?